File: examples/Rotating table/code_python/Supervisor/events.py
"""
Events from supremica file
"""
from . import Event, EventKind

# Create events
Events = {
    'a1': Event(EventKind.CONTROLLABLE, 0, 'a1'),
    'b1': Event(EventKind.UNCONTROLLABLE, 1, 'b1'),
    'a2': Event(EventKind.CONTROLLABLE, 2, 'a2'),
    'b2': Event(EventKind.UNCONTROLLABLE, 3, 'b2'),
    'se': Event(EventKind.UNCONTROLLABLE, 4, 'se'),
    'a0': Event(EventKind.CONTROLLABLE, 5, 'a0'),
    'b0': Event(EventKind.UNCONTROLLABLE, 6, 'b0'),
    'a3': Event(EventKind.CONTROLLABLE, 7, 'a3'),
    'a4': Event(EventKind.CONTROLLABLE, 8, 'a4'),
    'b3': Event(EventKind.UNCONTROLLABLE, 9, 'b3'),
    'b4': Event(EventKind.UNCONTROLLABLE, 10, 'b4'),
    'a5': Event(EventKind.CONTROLLABLE, 11, 'a5'),
    'b5': Event(EventKind.UNCONTROLLABLE, 12, 'b5'),
}

def get_event_by_id(event_id: int) -> Event:
    """
    Get event by id
    """
    if not isinstance(event_id, int):
        raise TypeError("event_id must be an integer")

    for event in Events.values():
        if event.get_id() == event_id:
            return event
    return None

File: examples/Rotating table/code_c/src/event_handler/events.c
#include <stdlib.h>
#include "events.h"
// create events
Event a1 = {CONTROLLABLE, 0, SUP_DEBUG_STR("a1"), NULL};
Event b1 = {UNCONTROLLABLE, 1, SUP_DEBUG_STR("b1"), NULL};
Event a2 = {CONTROLLABLE, 2, SUP_DEBUG_STR("a2"), NULL};
Event b2 = {UNCONTROLLABLE, 3, SUP_DEBUG_STR("b2"), NULL};
Event se = {UNCONTROLLABLE, 4, SUP_DEBUG_STR("se"), NULL};
Event a0 = {CONTROLLABLE, 5, SUP_DEBUG_STR("a0"), NULL};
Event b0 = {UNCONTROLLABLE, 6, SUP_DEBUG_STR("b0"), NULL};
Event a3 = {CONTROLLABLE, 7, SUP_DEBUG_STR("a3"), NULL};
Event a4 = {CONTROLLABLE, 8, SUP_DEBUG_STR("a4"), NULL};
Event b3 = {UNCONTROLLABLE, 9, SUP_DEBUG_STR("b3"), NULL};
Event b4 = {UNCONTROLLABLE, 10, SUP_DEBUG_STR("b4"), NULL};

Event *controllable_event_list[CONTROLLABLE_EVENTS_COUNT] = {&a1,&a2,&a0,&a3,&a4};